util/devel/lookForBadCompCalls

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+"""Check Chapel compiler for 'inappropriate' function calls
+
+Check for exit routines (should call clean_exit instead).
+"""
+
+import os
+import sys
+import look_for_calls
+
+chplenv_dir = os.path.join(os.path.dirname(__file__), '..', 'chplenv')
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+from chpl_home_utils import get_chpl_home
+
+
+def main():
+    """Check compiler for "bad function calls"""
+    chpl_home = get_chpl_home()
+    compiler_dir = os.path.join(chpl_home, 'compiler')
+    frontend_dir = os.path.join(chpl_home, 'frontend')
+
+    # Check runtime for exit calls
+    exit_exclude_paths = ['compiler/util/misc.cpp',
+                          'frontend/include/chpl/resolution/ResolvedVisitor.h',
+                          'frontend/include/chpl/uast/AstNode.h',
```

Comment:
Are there actual calls to exit in here, or are these false positives? We use `cscope` to find calls and I'm wondering if it's falling short for C++ method calls since the original intent was to use this with C code.

Reply:
`misc.cpp` has actual calls to `exit`, as do the tests. The `ResolvedVisitor.h` and `AstNode.h` are false positives (since we have methods named `exit` in the frontend).

Reply:
I see. False positives from the tool (cscope and thus ours) being designed for C instead of C++ is a little unfortunate.

Doesn't seem worth renaming the enter/exit visitors just to keep this cleaner, but does mean some real calls could slip in.

Reply:
Probably nothing easy to do about this. Could maybe do some post-processing and exclude lines that look like a method call, but this still seems like a strict improvement to me.

Reply:
Yeah, fortunately the false positive is limited to a small number of files. I think it's reasonable to proceed with these files simply filtered out.